Lib/test/test_uuid.py

```diff
@@ -311,6 +311,26 @@ def test_getnode(self):
         node2 = self.uuid.getnode()
         self.assertEqual(node1, node2, '%012x != %012x' % (node1, node2))
 
+    # bpo-32502: UUID1 requires a 48-bit identifier, but hardware identifiers
+    # need not necessarily be 48 bits (e.g., EUI-64).
+    def test_uuid1_eui64(self):
+        # Reset any cached node value
+        self.uuid._node = None
+
+        # Confirm that uuid.getnode ignores hardware addresses larger than 48
+        # bits
+        bad_getter = lambda: 1 << 48
+        node = self.uuid.getnode(getters=[bad_getter])
+        self.assertTrue(0 < node < (1 << 48), '%012x' % node)
+
+        # Confirm that uuid1 doesn't fail when there's only a 64-bit hardware
+        # address
+        try:
+            self.uuid.uuid1(node=node)
+        except ValueError as e:
+            self.fail('uuid1 was given an invalid node ID')
```

Comment:
This one *is* a bit of a head scratcher.  Why does the comment mention "only a 64-bit hardware address" but you're passing in `node` which should be 48 bits?  It's not easy to follow the logic, so what's going on here?

Reply:
As above, I'm actually passing it in something that's 49 bits, which could happen in a 64-bit address space but not in a 48-bit address space.

Reply:
To clarify further: I'm having the "hardware" return a number that's from an address space larger than 48 bits. Running this test against `master` means that `getnode()` will return that value and this check will fail. On this branch, `getnode()` rejects the "hardware" address and properly constraints the value of `node`.